solidity/contracts/bridge/Wallets.sol

```diff
@@ -0,0 +1,287 @@
+        WalletState state;
+    }
+
+    event NewWalletRequested();
+
+    event NewWalletRegistered(
+        bytes32 indexed ecdsaWalletID,
+        bytes20 indexed walletPubKeyHash160
+    );
+
+    /// @notice Initializes state invariants.
+    /// @param registry ECDSA Wallet Registry reference
+    /// @dev Requirements:
+    ///      - ECDSA Wallet Registry address must not be initialized
+    function init(Data storage self, EcdsaWalletRegistry registry) external {
+        require(
+            address(self.registry) == address(0),
+            "ECDSA Wallet Registry address already set"
+        );
+
+        self.registry = registry;
+    }
+
+    /// @notice Sets the wallet creation period.
+    /// @param creationPeriod New value of the wallet creation period
+    function setCreationPeriod(Data storage self, uint32 creationPeriod)
+        external
+    {
+        self.creationPeriod = creationPeriod;
+    }
+
+    /// @notice Sets the minimum and maximum BTC balance parameters.
+    /// @param minBtcBalance New value of the minimum BTC balance
+    /// @param maxBtcBalance New value of the maximum BTC balance
+    /// @dev Requirements:
+    ///      - Minimum BTC balance must be lower than maximum BTC balance
+    function setBtcBalanceRange(
+        Data storage self,
+        uint64 minBtcBalance,
+        uint64 maxBtcBalance
+    ) external {
+        require(
+            minBtcBalance < maxBtcBalance,
+            "Minimum must be lower than the maximum"
+        );
+
+        self.minBtcBalance = minBtcBalance;
+        self.maxBtcBalance = maxBtcBalance;
+    }
+
+    /// @notice Requests creation of a new wallet. This function just
+    ///         forms a request and the creation process is performed
+    ///         asynchronously. Outcome of that process should be delivered
+    ///         using `registerNewWallet` function.
+    /// @param activeWalletMainUtxo Data of the active wallet's main UTXO, as
+    ///        currently known on the Ethereum chain.
+    /// @dev Requirements:
+    ///      - `activeWalletMainUtxo` components must point to the recent main
+    ///        UTXO of the given active wallet, as currently known on the
+    ///        Ethereum chain. If there is no active wallet at the moment, or
+    ///        the active wallet has no main UTXO, this parameter can be
+    ///        empty as it is ignored.
+    ///      - Wallet creation must not be in progress
+    ///      - If the active wallet is set, one of the following
+    ///        conditions must be true:
+    ///        - The active wallet BTC balance is above the minimum threshold
+    ///          and the active wallet is old enough, i.e. the creation period
+    ///           was elapsed since its creation time
+    ///        - The active wallet BTC balance is above the maximum threshold
+    ///        If the active wallet is not set at the moment, there is no
+    ///        additional conditions.
```

Comment:
We said earlier "If the active wallet is set, one of the following conditions must be true:". I think we can eliminate these two lines for clarity.

Reply:
Eliminated within https://github.com/keep-network/tbtc-v2/pull/172/commits/c69551d355d3887644857a4b62f6040298dcb3ab